src/network/networkd-link.c

```diff
@@ -3081,8 +3081,9 @@ static int link_configure(Link *link) {
                         return r;
         }
 
-        if (link->network->mtu > 0) {
-                r = link_set_mtu(link, link->network->mtu);
+        if (link->network->mtu > 0 ||
+            (link->network->min_mtu > 0 && link->network->min_mtu > link->mtu)) {
```

Comment:
hmm, not sure i follow, but why conditoinalize the right-hand side of `&&` only on min_mtu > 0? shouldn't it be this:

```
mtu = MAX(link->network->mtu, link->network->min_mtu);
if (mtu > link->mtu) {
        …
```

but i must admit I don't fully grok the intention here